src/components/project/ProjectsList.tsx

```diff
@@ -28,14 +28,19 @@ const ProjectsList = (props: ProjectsListProps) => {
 
   const { searchFilterText, isShowingInActiveProjects } = props.filters
 
-  const filteredProjects = searchFilterText ? props.projects.filter(project => {
+  // ? Search filter
+  let filteredProjects = searchFilterText ? props.projects.filter(project => {
     const { consultant, partner, client, details } = project
     const startDate = formatDate(details.startDate)
     const endDate = formatDate(details.endDate)
 
     return searchinize(`${consultant.name} ${consultant.type} ${startDate} ${endDate} ${partner && partner.name} ${client.name}`)
       .includes(searchFilterText.toLowerCase())
-  }) : props.projects
+  })
+    : props.projects
+
+  // ? Filter projects by active status
+  filteredProjects = isShowingInActiveProjects ? filteredProjects.filter(project => !project.details.isActive) : filteredProjects.filter(project => project.details.isActive)
```

Comment:
Op deze manier gaat het potentieel inderdaad langer duren omdat je tweemaal over de array loopt.
"Guard Clauses" is wat hier typisch gebruikt wordt om snel te short circuiten

voorbeeld:

```
if (hasNoFilter) {
    return props.projects;
}

const filteredProjects = props.projects.filter(project => {
    const { consultant, partner, client, details } = project

   if (!isShowingInActiveProjects && !project.details.isActive) {
       // --> Dit is de guard clause: short cirtcuit indien deze record er sowieso uit moet
       return false;
   }

   if (!searchFilterText) {
        // nog een guard: short circuit wanneer het er sowieso bij moet
         return true;
    }

    const startDate = formatDate(details.startDate)
    const endDate = formatDate(details.endDate)

    return searchinize(`${consultant.name} ${consultant.type} ${startDate} ${endDate} ${partner && partner.name} ${client.name}`)
      .includes(searchFilterText.toLowerCase())
```

Reply:
Maar dit is misschien op zich ook al premature optimization.
Looping duurt nu typisch wel lang in programmatie
Maar de kans is heel groot dat er zo goed als geen verschil in timing is tussen de huidige implementatie en mijn voorstel

Reply:
`   if (!isShowingInActiveProjects && !project.details.isActive) {
       // --> Dit is de guard clause: short cirtcuit indien deze record er sowieso uit moet
       return false;
   }`

We moeten toch 2 lijsten laten zien?
- Lijst met enkel actieve projecten (toggle off)
- Lijst met enkel inactieve projecten (toggle on)

Dus dit gaat niet werken als we bovenstaande functionaliteit willen, of ben ik fout?

Reply:
Is eerder voorbeeld code maar functioneel is het:
- standaard: toon enkel de actieve
- toggle "show inactive": toon ze allemaal

Reply:
Aah oke dan had ik het verkeerd begrepen, dus ik mag de actieve en inactieve projecten door elkaar laten zien als show inactive true is? Is toch niet overzichtelijk?